### PR TITLE
Colorado Child Tax Credit

### LINKED ISSUE #2707: Colorado Child Tax Credit
[Reference](https://advance.lexis.com/documentpage/?pdmfid=1000516&crid=2fe6fa8e-8df8-4d74-a340-c4fa9bc603d5&nodeid=ABPAACAACAABAACABN&nodepath=%2FROOT%2FABP%2FABPAAC%2FABPAACAAC%2FABPAACAACAAB%2FABPAACAACAABAAC%2FABPAACAACAABAACABN&level=6&haschildren=&populated=false&title=39-22-129.+Child+tax+credit+-+legislative+declaration+-+definitions.&config=014FJAAyNGJkY2Y4Zi1mNjgyLTRkN2YtYmE4OS03NTYzNzYzOTg0OGEKAFBvZENhdGFsb2d592qv2Kywlf8caKqYROP5&pddocfullpath=%2Fshared%2Fdocument%2Fstatutes-legislation%2Furn%3AcontentItem%3A633G-B1C3-GXJ9-30GW-00008-00&ecomp=7gf59kk&prid=35495ea2-09ee-41c5-8d09-4eb29004ac3a
)

### PR DESCRIPTION
**Pull request by qingqiuzhang**
Fixes #2707

### Summary
📱🛠️🌐

Add a new policy parameter `child_allowance` to the US tax and benefit system. Update the `household_net_income` function to include the child allowance in the calculation. Add tests and documentation for the new parameter and function.

> _`is_valid` checks_
> _refactored for clarity_
> _autumn leaves falling_

### Walkthrough
*  Add a new parameter `child_allowance.amount` to the `parameters.yaml` file, which sets the monthly amount of the child allowance benefit per child ()

### CODE REVIEW

File: policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/age_threshold.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+description: Colorado issues child tax credit for taxpayers with child under this age.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/age_threshold.yaml:1]:
```suggestion
description: Colorado issues child tax credit to taxpayers with children below this age threshold.
```

File: policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/joint.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/joint.yaml:1]: we will also need files for the other three filing statuses (separate, widow, head of household)

File: policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/head_of_household.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  label: Colorado child tax credit head of household filers rate
+  reference:
+    - title: C.R.S. 39-22-129. Child tax credit - legislative declaration - definitions.
+      href: https://advance.lexis.com/documentpage/?pdmfid=1000516&crid=4f86a852-765a-489a-b554-66f6e7d5b065&nodeid=ABPAACAACAABAACABN&nodepath=%2fROOT%2fABP%2fABPAAC%2fABPAACAAC%2fABPAACAACAAB%2fABPAACAACAABAAC%2fABPAACAACAABAACABN&level=6&haschildren=&populated=false&title=39-22-129.+Child+tax+credit+-+legislative+declaration+-+definitions.&config=014FJAAyNGJkY2Y4Zi1mNjgyLTRkN2YtYmE4OS03NTYzNzYzOTg0OGEKAFBvZENhdGFsb2d592qv2Kywlf8caKqYROP5&pddocfullpath=%2fshared%2fdocument%2fstatutes-legislation%2furn%3acontentItem%3a633G-B1C3-GXJ9-30GW-00008-00&ecomp=7gf59kk&prid=2fe6fa8e-8df8-4d74-a340-c4fa9bc603d5
+    - title: 2022 Colorado Child Tax Credit
+      href: https://tax.colorado.gov/sites/tax/files/documents/DR_0104CN_2022.pdf#page=1
+    - title: Colorado Individual Income Tax Filing Guide - Instructions for Select Credits from the DR 0104CR - Line 1 Child Tax Credit
+      href: https://tax.colorado.gov/sites/tax/files/documents/DR_104_Book_2022.pdf#page=16
+brackets:
+  - threshold:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/head_of_household.yaml:15]: should this be 0?

File: policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/age_threshold.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+description: Colorado issues child tax credit to taxpayers with children below this age threshold.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/age_threshold.yaml:1]:
```suggestion
description: Colorado issues child tax credit to taxpayers with children below this age.
```

File: policyengine_us/variables/gov/states/co/tax/income/credits/ctc/co_ctc.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+class co_ctc(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Colorado child tax credit"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        # C.R.S. 39-22-129. Child tax credit - legislative declaration - definitions.
+        "https://advance.lexis.com/documentpage/?pdmfid=1000516&crid=4f86a852-765a-489a-b554-66f6e7d5b065&nodeid=ABPAACAACAABAACABN&nodepath=%2fROOT%2fABP%2fABPAAC%2fABPAACAAC%2fABPAACAACAAB%2fABPAACAACAABAAC%2fABPAACAACAABAACABN&level=6&haschildren=&populated=false&title=39-22-129.+Child+tax+credit+-+legislative+declaration+-+definitions.&config=014FJAAyNGJkY2Y4Zi1mNjgyLTRkN2YtYmE4OS03NTYzNzYzOTg0OGEKAFBvZENhdGFsb2d592qv2Kywlf8caKqYROP5&pddocfullpath=%2fshared%2fdocument%2fstatutes-legislation%2furn%3acontentItem%3a633G-B1C3-GXJ9-30GW-00008-00&ecomp=7gf59kk&prid=2fe6fa8e-8df8-4d74-a340-c4fa9bc603d5",
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/co/tax/income/credits/ctc/co_ctc.py:12]: this link isn't working

casetext might be more stable?
https://casetext.com/statute/colorado-revised-statutes/title-39-taxation/specific-taxes/income-tax/article-22-income-tax/part-1-general/section-39-22-129-child-tax-credit-legislative-declaration-definitions-repeal

File: policyengine_us/variables/gov/states/co/tax/income/credits/ctc/co_ctc.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.co.tax.income.credits.ctc
+        filing_status = tax_unit("filing_status", period)
+        statuses = filing_status.possible_values
+        person = tax_unit.members
+        # The co ctc amount is based on agi, federal ctc amount and number of eligible children.
+        agi = tax_unit("adjusted_gross_income", period)
+        federal_ctc = tax_unit("ctc", period)
+
+        eligible_child = person("age", period) < p.age_threshold
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/co/tax/income/credits/ctc/co_ctc.py:29]: the legal code adds the condition of being eligible for the federal ctc

File: policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/rate/head_of_household.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+description: Colorado issues the child tax credit at these rates for head of household filers.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/rate/head_of_household.yaml:1]:
```suggestion
description: Colorado matches this fraction of the federal child tax credit for head of household filers, depending on federal adjusted gross income.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/rate/head_of_household.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+description: olorado matches this fraction of the federal child tax credit for head of household filers, depending on federal adjusted gross income.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/ctc/rate/head_of_household.yaml:1]:
```suggestion
description: Colorado matches this fraction of the federal child tax credit for head of household filers, depending on federal adjusted gross income.
```

**Pull request merged**